sale_coupon_advanced/models/sale_order_line.py

```diff
@@ -4,8 +4,37 @@
 
 
 class SaleOrderLine(models.Model):
+    """Extend to add forced_reward_line field and override unlink."""
+
     _inherit = "sale.order.line"
 
     forced_reward_line = fields.Boolean(
         help="Tech field to cleanup automatically created lines"
     )
+
+    def unlink(self):
+        """Override to unlink disc lines when reward ones are unlinked.
+
+        Also removes related sale.coupon.program records.
+        """
+        lines_to_remove = self.env['sale.order.line']
+        SaleCouponProgram = self.env[
+            'sale.coupon.program']
+        for line in self:
```

Comment:
```suggestion
        for line in self.filtered(lambda self: self.forced_reward_line):
```

Reply:
As it is now, it should take care of all rewards, not just forced ones. Or we only want to remove related discount for forced lines?

Reply:
Hm, its not that its the problem, it is functionality. I can only apply to to forced lines, but then it will gimp functionality. Is there a reason, user would want to keep non forced discount line, if related promotion line was removed? If not, then I see no reason to use it only for forced line, cause it could be useful in both cases.